src/Database/Relations/BelongsToMany.php

```diff
@@ -411,12 +404,17 @@ public function getRelatedIds($sessionKey = null)
     }
 
     /**
-     * Get the pivot models that are currently attached (taking conditions & scopes into account).
+     * Create a new query builder for the pivot table.
      *
-     * @return \Illuminate\Support\Collection
+     * @return \Illuminate\Database\Query\Builder
      */
-    protected function getCurrentlyAttachedPivots()
+    public function newPivotQuery()
     {
-        return $this->getQuery()->get();
+        $query = parent::newPivotQuery();
+
+        // add relation's conditions and scopes to the query
+        $this->addDefinedConstraintsToQuery($query);
+
+        return $query->join($this->related->getTable(), $this->relatedPivotKey, '=', $this->relatedKey);
```

Comment:
@mjauvin Is there a reason this join is included?

Reply:
Yes, that's how the Constraints are made to work with the related table.

Reply:
@mjauvin Isn't that already done in the parent call though? I'm concerned this is adding an unnecessary constraint to *all* pivot queries.

Reply:
newPivotQuery() only returns the pivot table query, in order to apply the constraints, the related table needs to be joined to the Pivot table.

Reply:
In Laravel, they don't have "conditions" and "scopes" when defining the relations, so this is why their `newPivotQuery()` does not need to join with the related table.

Reply:
@mjauvin Unless I'm mistaken, these constraints are already added through the RelationController at various points, so this is effectively adding them again and then adding a join.

Before these changes, there was no necessity to implement a join, or even to overwrite the `newPivotQuery` method - of course, this didn't cover the particular case where there's two relations with the same target model but with different conditions, but that implies it *was* working with the more likely scenario of one relation with the target model.

I'm just trying to make sure this doesn't introduce an unintended side effect.

Reply:
@bennothommo you mean in `modules/backend/behaviors/RelationController` ? That code ?
```
            /*  
             * Apply defined constraints
             */
            if ($sqlConditions = $this->getConfig('view[conditions]')) {
                $widget->bindEvent('list.extendQueryBefore', function ($query) use ($sqlConditions) {
                    $query->whereRaw($sqlConditions);
                });
            }
            elseif ($scopeMethod = $this->getConfig('view[scope]')) {
                $widget->bindEvent('list.extendQueryBefore', function ($query) use ($scopeMethod) {
                    $query->$scopeMethod($this->model);
                });
            }
            else {
                $widget->bindEvent('list.extendQueryBefore', function ($query) use ($widget) {
                    $this->relationObject->addDefinedConstraintsToQuery($query);
                    if ($widget->getSortColumn()) {
                        $query->getQuery()->orders = [];
                    }
                });
            }
```